Gems/EMotionFX/Code/EMotionFX/Source/Recorder.cpp

```diff
@@ -1421,7 +1421,10 @@ namespace EMotionFX
     // extract sorted active items
     void Recorder::ExtractNodeHistoryItems(const ActorInstanceData& actorInstanceData, float timeValue, bool sort, EValueType valueType, AZStd::vector<ExtractedNodeHistoryItem>* outItems, AZStd::vector<size_t>* outMap) const
     {
-        // clear the map array
+        // Clear the item array and map array 
+        outItems->clear();
+        outMap->clear();
+
         const size_t maxIndex = CalcMaxNodeHistoryTrackIndex(actorInstanceData);
         outItems->resize(maxIndex + 1);
         for (size_t i = 0; i <= maxIndex; ++i)
```

Comment:
This is clearing the `outItems` vector, destroying all the contained objects, then resizing it back up again, which default constructs a bunch of `ExtractedNodeHistoryItem`s, then looping to explicitly set the values for each element to 0, which is already done by the default construction.
* You shouldn't need to call `outItems->clear()` here, because of the `resize()` call that happens immediately afterwards
* You should be able to remove the following `for` loop, it doesn't really do anything

Where was this crashing? I'm not understanding how this would prevent a crash, it seems to me that regardless of if you call `outItems->clear()` and `outMap->clear()`, at the end of this function, their sizes will be `maxIndex + 1`.